### PR TITLE
Fix mssql-plugin occur ClassCastException when call the method of return generate key

### PR DESCRIPTION
**Pull request by zifeihan**
### Fix mssql-plugin occur ClassCastException when call the method of return generate key
- [ ] Add a unit test to verify that the fix works.
- [x] Explain briefly why the bug exists and how to fix it.
     
fix https://github.com/apache/skywalking/issues/6080, 
Previously, I wrapped the return value of the prepareStatement method of SWPreparedStatement type, but in the method `com.microsoft.sqlserver.jdbc.SQLServerConnection#prepareStatement(java.lang.String, int, com.microsoft.sqlserver.jdbc.SQLServerStatementColumnEncryptionSetting)` , Type conversion, a `java.lang.ClassCastException: org.apache.skywalking.apm.plugin.jdbc.trace.SWPreparedStatement cannot be cast to com.microsoft.sqlserver.jdbc.SQLServerPreparedStatement`, fix method: using the interceptor to replace wrapper.

### CODE REVIEW

File: apm-sniffer/apm-sdk-plugin/mssql-jdbc-plugin/src/main/resources/skywalking-plugin.def [hunk 1]
```diff
@@ -16,3 +16,5 @@
 
 mssql-jdbc=org.apache.skywalking.apm.plugin.mssql.jdbc.define.DriverInstrumentation
 mssql-jdbc=org.apache.skywalking.apm.plugin.mssql.jdbc.define.ConnectionInstrumentation
+mssql-jdbc=org.apache.skywalking.apm.plugin.mssql.jdbc.define.PreparedStatementInstrumentation
+mssql-jdbc=org.apache.skywalking.apm.plugin.mssql.jdbc.define.StatementInstrumentation
```
Comment (wu-sheng) [inline apm-sniffer/apm-sdk-plugin/mssql-jdbc-plugin/src/main/resources/skywalking-plugin.def:20]: mssql plugin doesn't instrument statement before? I am a little confused.

**Comment by wu-sheng** [comment]
You need to explain what you are doing and why you are doing in the PR, please fix this. 
From my understanding, your chance is using the interceptor to replace wrapper, if so, why these interceptors are not in the jdbc-common, instead, you created msql-common?

File: apm-sniffer/apm-sdk-plugin/mssql-jdbc-plugin/src/main/resources/skywalking-plugin.def (hunk 1 above)
Comment (zifeihan) [reply apm-sniffer/apm-sdk-plugin/mssql-jdbc-plugin/src/main/resources/skywalking-plugin.def:20]: In the previous packaging mode, statement was packaged by `org.apache.skywalking.apm.plugin.jdbc.JDBCStatementInterceptor`.

**Comment by zifeihan** [comment]
> You need to explain what you are doing and why you are doing in the PR, please fix this.
> From my understanding, your chance is using the interceptor to replace wrapper, if so, why these interceptors are not in the jdbc-common, instead, you created msql-common?

```
    public PreparedStatement prepareStatement(String sql,
            int flag,
            SQLServerStatementColumnEncryptionSetting stmtColEncSetting) throws SQLServerException {
        loggerExternal.entering(getClassNameLogging(), "prepareStatement", new Object[] {sql, new Integer(flag), stmtColEncSetting});
        checkClosed();
        SQLServerPreparedStatement ps = (SQLServerPreparedStatement) prepareStatement(sql, ResultSet.TYPE_FORWARD_ONLY, ResultSet.CONCUR_READ_ONLY,
                stmtColEncSetting);
        ps.bRequestedGeneratedKeys = (flag == Statement.RETURN_GENERATED_KEYS);
        loggerExternal.exiting(getClassNameLogging(), "prepareStatement", ps);
        return ps;
    }
```
For example, in this method com.microsoft.sqlserver.jdbc.SQLServerConnection#prepareStatement(java.lang.String, int, com.microsoft.sqlserver.jdbc.SQLServerStatementColumnEncryptionSetting), the return value of the prepareStatement method is forcibly converted. Because mssql-jtds and mssql-jdbc have certain similarities, a new mssql-commons module has been added.

**Comment by wu-sheng** [comment]
You forget to update the changelog. Please fix.

**Review by wu-sheng — DISMISSED** [review]

**Review by wu-sheng — APPROVED** [review]
Once the plugin test passed, it is good to me, as the expected data isn't changed.

**Pull request merged**